src/Stratis.Bitcoin/P2P/PeerConnector.cs

```diff
@@ -282,6 +282,7 @@ public virtual TimeSpan CalculateConnectionInterval()
         /// <param name="peer">Peer that is being disposed.</param>
         private void OnPeerDisposed(INetworkPeer peer)
         {
+            this.connectionManager.Counter.Add(peer.Counter);
```

Comment:
I don't think this is the right place to do it, this only applies to outbounds, we should use do this where we modify the collection `connectedPeers`

Reply:
Agreed